cmd/packaging/windows-msi.go

```diff
@@ -129,34 +203,41 @@ func windowsMsiProcessFiles(path string) {
 
 	for _, f := range files {
 		p := filepath.Join(path, f.Name())
-		relativePath := strings.Split(strings.Split(p, "flutter_assets"+string(filepath.Separator))[1], string(filepath.Separator))
+		relativePath := strings.Split(strings.Split(p, "flutter_assets"+fileSeparator)[1], fileSeparator)
 		if f.IsDir() {
 			directoriesFileContent = append(directoriesFileContent,
-				`<Directory Id="FLUTTERASSETSDIRECTORY_`+strings.Join(relativePath, "_")+`" Name="`+f.Name()+`">`,
+				fmt.Sprintf("<Directory Id=\"FLUTTERASSETSDIRECTORY_%s\" Name=\"%s\">", hashSha1(strings.ReplaceAll(strings.ReplaceAll(strings.Join(relativePath, "_"), " ", "_"), "-", "_")), f.Name()),
 			)
-			windowsMsiProcessFiles(p)
+			windowsMsiProcessFiles(p, fileSeparator)
 			directoriesFileContent = append(directoriesFileContent,
-				`</Directory>`,
+				"</Directory>",
 			)
 		} else {
 			if len(relativePath) > 1 {
 				directoryRefsFileContent = append(directoryRefsFileContent,
-					`<DirectoryRef Id="FLUTTERASSETSDIRECTORY_`+strings.Join(relativePath[:len(relativePath)-1], "_")+`">`,
+					fmt.Sprintf("<DirectoryRef Id=\"FLUTTERASSETSDIRECTORY_%s\">", hashSha1(strings.ReplaceAll(strings.ReplaceAll(strings.Join(relativePath[:len(relativePath)-1], "_"), " ", "_"), "-", "_"))),
 				)
 			} else {
 				directoryRefsFileContent = append(directoryRefsFileContent,
-					`<DirectoryRef Id="FLUTTERASSETSDIRECTORY">`,
+					"<DirectoryRef Id=\"FLUTTERASSETSDIRECTORY\">",
 				)
 			}
 			directoryRefsFileContent = append(directoryRefsFileContent,
-				`<Component Id="flutter_assets_`+strings.Join(relativePath, "_")+`" Guid="*">`,
-				`<File Id="flutter_assets_`+strings.Join(relativePath, "_")+`" Source="build/flutter_assets/`+strings.Join(relativePath, "/")+`" KeyPath="yes"/>`,
-				`</Component>`,
-				`</DirectoryRef>`,
+				fmt.Sprintf("<Component Id=\"flutter_assets_%s\" Guid=\"*\">", hashSha1(strings.ReplaceAll(strings.ReplaceAll(strings.Join(relativePath, "_"), " ", "_"), "-", "_"))),
+				fmt.Sprintf("<File Id=\"flutter_assets_%s\" Source=\"build%sflutter_assets%s%s\" KeyPath=\"yes\"/>", hashSha1(strings.ReplaceAll(strings.ReplaceAll(strings.Join(relativePath, "_"), " ", "_"), "-", "_")), fileSeparator, fileSeparator, strings.Join(relativePath, fileSeparator)),
+				"</Component>",
+				"</DirectoryRef>",
 			)
 			componentRefsFileContent = append(componentRefsFileContent,
-				`<ComponentRef Id="flutter_assets_`+strings.Join(relativePath, "_")+`"/>`,
+				fmt.Sprintf("<ComponentRef Id=\"flutter_assets_%s\"/>", hashSha1(strings.ReplaceAll(strings.ReplaceAll(strings.Join(relativePath, "_"), " ", "_"), "-", "_"))),
 			)
 		}
 	}
 }
+
+func hashSha1(content string) string {
+	h := sha1.New()
+	h.Write([]byte(content))
+	sha := h.Sum(nil)
```

Comment:
These three lines could be simplified to:
```
contentSha := sha1.Sum([]byte(content))
```

Reply:
That code didn't work. Don't know why, but I'll just leave it as it is